python/mlx/optimizers.py

```diff
@@ -2,10 +2,18 @@
 
 import math
 from typing import List
-
+import sys
+import inspect
 import mlx.core as mx
 from mlx.utils import tree_map
 
+def get_all_optimizers(): # a auxiliar function to get all the optimizers, for testing purposes
+    classes = dict()
+    for name, obj in inspect.getmembers(sys.modules[__name__]):
+        if inspect.isclass(obj):
+            if obj.__name__ not in ["OptimizerState", "Optimizer"]:
+                classes[name] = obj
+    return classes
```

Comment:
Could we put this in the test file? It seems like a better fit there as it doesn't need to be in the package itself.

I think if you change to `inspect.getmembers(optim)` it should work.